db/db.go

```diff
@@ -0,0 +1,75 @@
+package db
+
+/*
+
+client<->entity relationship:
+
+get/set/delete accept the Path() argument from entity to perform key/value
+lookup. Get and set respectively manage Read() and Write() from entity.
+*/
+
+// Client provides a method of hooking into our data stores. It processes
+// entities (see below) and their pipeline between them and k/v stores such as
+// etcd and consul.
+type Client interface {
+	// Get the entity. Will copy the contents of the JSON into the entity for
+	// use.
+	Get(Entity) error
+	// Set the entity. Will persist the entity to etcd based on the path it
+	// provides. (See Path())
+	Set(Entity) error
+	// Delete the entity from the database. Does nothing to the underlying
+	// Object.
+	Delete(Entity) error
+	// Watch the Entity; just the entity and nothing else. Returns a channel for
+	// Entity updates and one for errors.
+	Watch(Entity) (chan Entity, chan error) // watch of entity
+	// WatchStop stops a watch on a specific Entity.
+	WatchStop(Entity) error
+	// WatchAll watches the prefix of the entity for changes, watching all of the
+	// specific type.
+	WatchAll(Entity) (chan Entity, chan error) // watch of subtree
```

Comment:
How about WatchTree?

On 15 Aug 2016, at 22:11, Vikrant Balyan wrote:

>> +	// use.
>> +	Get(Entity) error
>> +	// Set the entity. Will persist the entity to etcd based on the 
>> path it
>> +	// provides. (See Path())
>> +	Set(Entity) error
>> +	// Delete the entity from the database. Does nothing to the 
>> underlying
>> +	// Object.
>> +	Delete(Entity) error
>> +	// Watch the Entity; just the entity and nothing else. Returns a 
>> channel for
>> +	// Entity updates and one for errors.
>> +	Watch(Entity) (chan Entity, chan error) // watch of entity
>> +	// WatchStop stops a watch on a specific Entity.
>> +	WatchStop(Entity) error
>> +	// WatchAll watches the prefix of the entity for changes, watching 
>> all of the
>> +	// specific type.
>> +	WatchAll(Entity) (chan Entity, chan error) // watch of subtree
>
> WatchAll -> WatchHierarchy
> I prefer it so, but really up to you.. does not bother me much..
>
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/408/files/1a3157f1b1181a25f0c3c2e5792ffecb6ca6e07d#r74876893